refreshable/src/main/scala/prometheus4cats/refreshable/InstrumentedRefreshable.scala

```diff
@@ -0,0 +1,256 @@
+package prometheus4cats.refreshable
+
+import cats.effect.kernel.{MonadCancelThrow, Resource}
+import cats.effect.kernel.syntax.monadCancel._
+import cats.effect.kernel.syntax.resource._
+import cats.syntax.flatMap._
+import cats.syntax.foldable._
+import cats.syntax.functor._
+import cats.{Applicative, Monad}
+import com.permutive.refreshable.{CachedValue, Refreshable}
+import prometheus4cats._
+import retry.RetryDetails
+
+import scala.concurrent.duration.FiniteDuration
+
+class InstrumentedRefreshable[F[_], A] private (
+    underlying: Refreshable[F, A],
+    name: String,
+    readCounter: Counter.Labelled[F, Long, (String, CachedValue[A])],
+    runningGauge: Gauge.Labelled[F, Boolean, String],
+    exhaustedRetriesGauge: Gauge.Labelled[F, Boolean, String]
+)(implicit
+    override val functor: Monad[F]
+) extends Refreshable[F, A] {
+  override def get: F[CachedValue[A]] =
+    underlying.get.flatTap(value => readCounter.inc(name -> value))
```

Comment:
These should probably all be
```
F.uncancelable { poll =>
  poll(underlying).flatTap(metrics)
}
```
to ensure we always update metrics if the underlying operation is completed